index.js

```diff
@@ -2575,13 +2668,15 @@
 
     //  wrapNext :: (TypeVarMap, Array Any, Integer) -> (a -> b)
     const wrapNext = (_typeVarMap, _values, index) => (head, ...tail) => {
+      if (!$.checkTypes) {
+        return impl (head, ...tail);
+      }
```

Comment:
Here we check the current value of `$.checkTypes` to determine whether to perform type checking.